Source/WTF/wtf/StackTrace.cpp

```diff
@@ -45,6 +45,55 @@ void WTFGetBacktrace(void** stack, int* size)
 
 namespace WTF {
 
+#if USE(LIBBACKTRACE)
+static struct backtrace_state* backtraceState()
+{
+    static struct backtrace_state* backtraceState = backtrace_create_state(nullptr, 1, nullptr, nullptr);
+    return backtraceState;
+}
+
+static void backtraceSyminfoCallback(void* data, uintptr_t, const char* symname, uintptr_t, uintptr_t)
+{
+    const char** symbol = static_cast<const char**>(data);
+    *symbol = symname;
+}
+
+static int backtraceFullCallback(void* data, uintptr_t, const char*, int, const char* function)
+{
+    const char** symbol = static_cast<const char**>(data);
+    *symbol = function;
+    return 0;
+}
+
+const char** symbolize(void* const* addresses, int size)
```

Comment:
I see you're matching a preexisting interface here, but returning memory via raw pointer is pretty bad because it's easy to mess up. In this case, I see the return value is "transfer container" in that the array of strings but not the strings themselves need to be freed, and it will be freed once using `free` at the bottom of `StackTraceSymbolResolver`. Alas, that means the symbols returned by `abi::__cxa_demangle` are leaked. So we really need to return `char**` rather than `const char**` and each string in the array will need to be freed as well.

A better solution would be to convert everything to use `UniquePtr` so the caller doesn't have to worry about how to free it.

Reply:
You're right. It was a leak. I fixed it by freeing every string. Using `UniquePtr` makes the code much more verbose. I have an idea how we can refactor `StackTraceSymbolResolver` but I want to land this first.